File: shopapp-angular-prefix/shopapp-angular/src/app/dtos/order/order.dto.ts
import {
  IsString,
  IsPhoneNumber, IsNotEmpty, IsNumber, ArrayMinSize, ValidateNested
} from 'class-validator';
import {Product} from "../../models/product";
import {CartItemDto} from "./cart.item.dto";
import {Type} from "class-transformer";

export class OrderDto {
  user_id: number;
  fullname: string;
  email: string;
  phone_number: string;
  address: string;
  note: string;
  total_money: number;
  payment_method: string;
  shipping_method: string;
  coupon_code: string;
  shipping_address: string;
  cart_items: {product_id: number, quantity: number}[];
  constructor(data: any) {
    this.shipping_address = data.shipping_address;
    this.address = data.address;
    this.user_id = data.user_id;
    this.fullname = data.fullname;
    this.shipping_method = data.shipping_method;
    this.coupon_code = data.coupon_code;
    this.payment_method = data.payment_method;
    this.phone_number = data.phone_number;
    this.email = data.email;
    this.total_money = data.total_money;
    this.note = data.note;
    this.cart_items = data.cart_items;
  }
}

File: shopapp-backend/shopapp/src/main/java/com/project/shopapp/dtos/OrderDTO.java
package com.project.shopapp.dtos;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.*;
import org.springframework.stereotype.Component;

import java.time.LocalDate;
import java.util.Date;
import java.util.List;
import java.util.Map;

@Getter
@Setter
@Data
@Builder
@Component
@AllArgsConstructor
@NoArgsConstructor
public class OrderDTO {
    @JsonProperty("user_id")
    @Min(value = 1,message = "User's ID must be greater than 0")
    private Long userId;
    @JsonProperty("fullname")
    @NotBlank(message = "You must type down your name")
    private String fullName;

    private String email;

    @JsonProperty("phone_number")
    @NotBlank(message = "Please enter your phone number")
    @Size(min = 5,message = "Phone number size must be greater than 5")
    private String phoneNumber;

    @NotBlank(message = "Please enter your address to receive package")
    private String address;

    private String note;
    @JsonProperty("total_money")
    @Min(value = 0,message = "Total money must be greater than 0")
    private Float totalMoney;

    @JsonProperty("shipping_method")
    @NotBlank(message = "Please enter your shipping method")
    private String shippingMethod;

    @JsonProperty("shipping_address")
    @NotBlank(message = "Please enter your shipping address")
    private String shippingAddress;

    @JsonProperty("shipping_date")
    private LocalDate shippingDate;
    @JsonProperty("payment_method")
    @NotBlank(message = "Please choose one payment method")
    private String paymentMethod;

    @JsonProperty("cart_items")
    private List<CartItemDTO> cartItems;
}
